### PR TITLE
Resubmmited alternate PR for Pepperspray rework, Fucked up my repo

### PR DESCRIPTION
**Pull request by MacBlaze1**
PR machine broke, had to resubmit

close #46269
## About The Pull Request

Volume increased to 50 for pepper spray, added knockdown of decreasing degree for decreasing importance pepper spray entrance organs, added confuse, blind, blur, slow down. Adds a new clothing flag for head and mouth, PEPPERPROOF, that will decide if you can be pepper-sprayed while wearing those. Decreases scope of pepperspray protection by a lot.
## Why It's Good For The Game

Pepper spray is not even worth using right now, and I routinely see sec officers throwing away their pepper spray for e-bolas, handcuffs or flashbangs. This PR will make pepper spray a viable option in the sec arsenal, but does account for the range and spam ability of pepper spray while still making it competitive. As we are moving away from hardstuns, I see it fit to replace the hard stun with some bad effects, slowdown and knockdown. Helmets, Hardsuits, Gas masks, and other things like that are pepperproof
## Changelog
:cl:
balance: increased volume of pepper sprays to 50
tweak: adds knockdown of 3 second for no protection, 3 for no mouth protection
tweak: Changes bluring and blindness of eyes
balance: Replaces Paralysis with blind
balance: vastly decreased the scope of pepperspray protection.
add: New variable for pepperspray protection, PEPPERPROOF, decides if you can be pepperproof with certain equipment
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 1]
```diff
@@ -352,13 +352,6 @@
 	color = "#302000" // rgb: 48, 32, 0
 	taste_description = "bitterness"
 
-/datum/reagent/consumable/coco/on_mob_add(mob/living/carbon/M)
-	.=..()
-	if(iscatperson(M))
-		to_chat(M, "<span class='warning'>Your insides revolt at the presence of lethal chocolate!</span>")
-		M.vomit(20)
```
Comment (nemvar) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: beep boop, conflict not resolved properly

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 2]
```diff
@@ -283,33 +283,33 @@
...
-			victim.blur_eyes(3)
-			victim.blind_eyes(2)
-			victim.confused = max(M.confused, 3)
-			victim.damageoverlaytemp = 60
-			victim.Paralyze(60)
+			victim.blur_eyes(3) // 6 seconds
+			victim.blind_eyes(3) // 6 seconds
+			victim.confused = max(M.confused, 5)
 			return
 		else if ( eyes_covered ) // Eye cover is better than mouth cover
```
Comment (nemvar) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: Update the comment.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 3]
```diff
@@ -283,33 +283,33 @@
 	var/mob/living/carbon/victim = M
 	if(method == TOUCH || method == VAPOR)
 		//check for protection
-		var/mouth_covered = victim.is_mouth_covered()
 		var/eyes_covered = victim.is_eyes_covered()
```
Comment (nemvar) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: is_eyes_covered returns whatever thing covers the eyes first, including gas masks.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 4]
```diff
@@ -283,33 +283,33 @@
...
 
 		//actually handle the pepperspray effects
-		if ( eyes_covered && mouth_covered )
+		if (eyes_covered && istype(victim.wear_mask, /obj/item/clothing/mask/gas))
```
Comment (nemvar) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: This actually results gas masks being better than sunglasses against pepperspray. If I read the description correctly, that was not the intent?

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 5]
```diff
@@ -283,33 +283,33 @@
...
+			victim.confused = max(M.confused, 5)
 			return
 		else if ( eyes_covered ) // Eye cover is better than mouth cover
-			victim.blur_eyes(3)
-			victim.damageoverlaytemp = 30
+			victim.blur_eyes(2) // 4 seconds
+			victim.Knockdown(3 SECONDS)
+			victim.add_movespeed_modifier(MOVESPEED_ID_PEPPER_SPRAY, update=TRUE, priority=100, multiplicative_slowdown=0.25, blacklisted_movetypes=(FLYING|FLOATING))
+			addtimer(CALLBACK(victim, /mob.proc/remove_movespeed_modifier, MOVESPEED_ID_PEPPER_SPRAY), 10 SECONDS)
 			return
```
Comment (nemvar) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: I'm usually all for returning as soon as possible but you can get rid of three lines of duplicate code if you don't.
```
			victim.Knockdown(3 SECONDS)
			victim.add_movespeed_modifier(MOVESPEED_ID_PEPPER_SPRAY, update=TRUE, priority=100, multiplicative_slowdown=0.25, blacklisted_movetypes=(FLYING|FLOATING))
			addtimer(CALLBACK(victim, /mob.proc/remove_movespeed_modifier, MOVESPEED_ID_PEPPER_SPRAY), 10 SECONDS)
```
are shared behaviour between this case and the next.

Comment (MacBlaze1) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Ah, I was combining the effects of the two.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm (hunk 4 above)
Comment (MacBlaze1) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Yeah I dont know, I tried to balance the effects, but gas masks are the only mouth protection with eye protection against pepper spray.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm (hunk 3 above)
Comment (MacBlaze1) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: What would be a better thing? Checking if there is an item in their eye slot?

**Comment by ATH1909** [comment]
"Gas masks and only gas masks block pepper spray, this is an exact parallel to flashes and sunglasses."
This is blatantly untrue; you should probably remove this from the changelog and your PR's opening post.

**Comment by ATH1909** [comment]
Here's an idea: instead of checking to see if someone's mouth is covered, check to see if they're wearing a gas mask (or a gas mask-like item) OR if they have their internals on (and don't have the TRAIT_NOBREATH trait).

In other words, pepper spray would only NOT work on someone if all of the following conditions are met:
- Their eyes are covered, whether by a (gas) mask, science goggles, or something else, or they don't have eyes/don't need eyes.
- They either have their internals on (as in, they're breathing from a gas tank or the like), have the TRAIT_NOBREATH trait, or are wearing a gas mask (or a gas mask-like item, like a sechailer or a SWAT mask).
- They're not a carbon.

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 6]
```diff
@@ -23,15 +23,25 @@
...
+/mob/living/carbon/is_mouth_pepper_proof(check_head = TRUE, check_mask = TRUE)
+	if(check_head && head && (head.flags_cover & HEADPEPPERPROOF))
+		return head
+	if(check_mask && wear_mask && (wear_mask.flags_cover & MASKPEPPERPROOF))
+		return wear_mask
+/mob/living/carbon/is_eyes_pepper_proof(check_head = TRUE, check_glasses = TRUE)
+	if(check_head && head && (head.flags_cover & HEADPEPPERPROOF))
+		return head
+	if(check_glasses && glasses && (glasses.flags_cover & EYESPEPPERPROOF))
+		return glasses
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/carbon_defense.dm]: all of these should be

`check_x && x?.flags_cover & y`

File: code/__DEFINES/inventory.dm [hunk 7]
```diff
@@ -156,7 +156,9 @@
 #define HEADCOVERSEYES		(1<<2)		// feel free to realloc these numbers for other purposes
 #define MASKCOVERSMOUTH		(1<<3)		// on other items, these are just for mask/head
 #define HEADCOVERSMOUTH		(1<<4)
-
+#define MASKPEPPERPROOF		(1<<5)	//protects against pepperspray 
+#define HEADPEPPERPROOF		(1<<6)
+#define EYESPEPPERPROOF		(1<<7)
```
Comment (ExcessiveUseOfCobblestone) [inline code/__DEFINES/inventory.dm]: this should really just be 1 flag, i'm not sure why each clothing piece uses a unique bit.

**Comment by ATH1909** [comment]
Did you check to make sure that explorer gas masks will be properly-pepper-proof?

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 8]
```diff
@@ -23,15 +23,20 @@
...
 /mob/living/carbon/is_eyes_covered(check_glasses = TRUE, check_head = TRUE, check_mask = TRUE)
 	if(check_head && head && (head.flags_cover & HEADCOVERSEYES))
 		return head
 	if(check_mask && wear_mask && (wear_mask.flags_cover & MASKCOVERSEYES))
 		return wear_mask
 	if(check_glasses && glasses && (glasses.flags_cover & GLASSESCOVERSEYES))
 		return glasses
-
+/mob/living/carbon/is_pepper_proof(check_head = TRUE, check_mask = TRUE)
+	if(check_head &&(head.flags_cover & PEPPERPROOF))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
	if(check_head &&(head?.flags_cover & PEPPERPROOF))
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 9]
```diff
@@ -23,15 +23,20 @@
...
+		return head
+	if(check_mask &&(wear_mask.flags_cover & PEPPERPROOF))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
	if(check_mask &&(wear_mask?.flags_cover & PEPPERPROOF))
```

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]

**Comment by MacBlaze1** [comment]
> Did you check to make sure that explorer gas masks will be properly-pepper-proof?

@ATH1909 yes, they are a subtype of gas mask

File: code/modules/mob/living/living_defense.dm [hunk 10]
```diff
@@ -38,7 +38,8 @@
 
 /mob/living/proc/is_eyes_covered(check_glasses = 1, check_head = 1, check_mask = 1)
 	return FALSE
-
+/mob/living/proc/is_pepper_proof(check_head = 1, check_mask = 1)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/living_defense.dm]:
```suggestion
/mob/living/proc/is_pepper_proof(check_head = TRUE, check_mask = TRUE)
```

**Comment by ExcessiveUseOfCobblestone** [comment]
r2m after

**Pull request merged**